docs/jegyzet/ado_net/ADO_NET.md

```diff
@@ -0,0 +1,275 @@
+
+Az adatbázissal való kapcsolat létrehozásához szükségünk van egy az előbb említett **Connection string**-re! Ez a szöveges változó írja le, hogy milyen paraméterekkel szeretnénk az adatbázishoz csatlakozni! Ilyen például a felhasználónév, jelszó, vagy éppen a szerver címe! 
+
+> A **Connection string**-eknek adatbázis szerver típusonként eltérő szintaktikájuk lehet, és [veszélyforrásokat](#connection-string) is rejthetnek magukban. 
+
+**Példa** kapcsolat létrehozására `StringBuilder`-rel:
+```cs
+var builder = new SqlConnectionStringBuilder();
+builder.UserID = "User";
+builder.Password = "Pw";
+builder.DataSource = "database.server.hu";
+builder.InitialCatalog = "Adatvez";
+
+var conn = new SqlConnection(builder.ConnectionString);
+conn.Open();
+... //lekérdezések, eljárások végrehajtása
+conn.Close(); //nem felejtjük el bezárni, hogy elkerüljük a Connection leak-et
+```
+
+## Command ##
+Az adatbázis kapcsolat létrehozása után szeretnénk az adatbázissal kommunikálni, lekérdezéseket, eljárásokat futtatni. Ehhez az **ADO** az `IDbCommand` interfészt biztosítja, ami egy utasítást reprezentál.
+
+### Előkészítés ###
+Az `IDbCommand` alábbi főbb *property*-jeit beállítva tudjuk megmondani, hogy a megadott parancs, hogyan legyen értelmezve, és miként viselkedjen az utasítás:
+- `CommandType`:  háromféle lehet
+-- Tárolt eljárás (*StoredProcedure*)
+-- Tábla teljes tartalmához hozzáférés (*TableDirect, nem minden implementáció használja, pl. OleDB igen*)
+-- SQL query (*Text*) - alapértelmezett
+- `CommandText`: maga a parancs szövege
+- `Connection`: az adatbázis kapcsolat
+- `Transaction`: a tranzakció
+- `CommandTimeout`: timeout arra az esetre, ha valami elakad (*alapértelmezetten 30s*)
+- `Parameters`: paraméterek, az [SQL injection támadás](#veszelyforrasok) kivédéséhez
+
+### Végrehajtás ###
+Miután előkészítettük a `Command`-ot, végre is hajthatjuk azt. Ehhez többféle lehetőség közül is választhatunk, attól függően, hogy mit várunk visszatérési értéknek.
+
+- `ExecuteReader`: több rekord lekérdezése 
+- `ExecuteScalar`: skalár érték lekérdezése
+- `ExecuteNonQuery`: olyan lekérdezés, amit nem eredményhalmazt ad vissza (pl. INSERT) - az érintett sorok számát adja helyette vissza
+- `ExecuteXmlReader` (MS SQL Server): XML dokumentumot (`XmlReader`* objektum*) ad vissza, az eredmény egy rekord egyetlen XML oszlopa 
+
+Parancsokat újrafelhasználni is tudunk, a `Command.Prepare()` (*szerver oldalon előkészíti a parancs futtatását*) hívás után, ám ez csak akkor éri meg, ha elegendően sokszor futtatjuk az utasítást, mivel az előkészítés és előkészülten tartás idő- és erőforrásigényes. 
+
+**Példa** parancs használatára:
+```cs
+//kapcsolatot létrehozása, megnyitása
+...
+var command = new SqlCommand();

```

Comment:
C#-ban szeretem az object initializer syntax-ot

```cs
var command = new SqlCommand()
{
    Connection = connection,
    CommandType = CommandType.StoredProcedure,
    CommandText = "salesByCategory", //StoredProcedure, ezért elég a tárolt eljárás nevét megadni
}
```

Reply:
De egyébként a konstruktornak is vannak talán olyan overloadjai, aminek ezeket meg lehet adni. Igazából a tananyag szempontjából ez nem lényeges, a mostani kód is szemléletes

Reply:
beletettem